src/Raven.Embedded/RavenEmbedded.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Text;
+using Raven.Client.Documents;
+using Raven.Client.Exceptions;
+using Raven.Client.Extensions;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Sparrow.Logging;
+
+
+namespace Raven.Embedded
+{
+    public class RavenEmbedded
+    {
+        private Process _ravenDbSlavedProcess;
+        private readonly Logger Logger = LoggingSource.Instance.GetLogger<RavenEmbedded>("Embedded");
+        private Task<Uri> _serverTask;
+        private long _initialized;
+
+        private readonly ConcurrentDictionary<string, IDocumentStore> _documentStores = new ConcurrentDictionary<string, IDocumentStore>();
+
+        public void StartServer(ServerOptions options = null)
+        {
+            // Start the server
+
+            if (Interlocked.CompareExchange(ref _initialized, 1, 0) != 0)
+                throw new InvalidOperationException("Already have been initialized");
+
+            options = options ?? ServerOptions.Default;
+            _serverTask = RunServer(options);
+        }
+
+        public static RavenEmbedded Instance = new RavenEmbedded();
+
+        public async Task<IDocumentStore> GetDocumentStore(string database)
+        {
+            if (Interlocked.Read(ref _initialized) == 0)
+                throw new InvalidOperationException("Please run StartServer before trying to get document store");
+
+            if (string.IsNullOrEmpty(database))
+                throw new ArgumentException(nameof(database));
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"GetDocumentStore for db ${ database }.");
+
+            var serverUrl = await GetServerUri();
+
+            if (_documentStores.TryGetValue(database, out var existingStore))
+            {
+                return existingStore;
+            }
+
+            var store = new DocumentStore
+            {
+                Urls = new[] { serverUrl.AbsoluteUri },
+                Database = database
+
+            }.Initialize();
+
+            try
+            {
+                await store.Maintenance.Server.SendAsync(new CreateDatabaseOperation(new DatabaseRecord(database))).ConfigureAwait(false);
```

Comment:
Need a way for the user to configure the `DatabaseRecord` here.

Reply:
@ayende We want to do it in a 2nd phase (2nd PR).

@IdanHaim check how this is done in `TestBase`.

Reply:
I have an Idea how to do so.